OEP-54: Core Contributor Program (#275)
openedx/open-edx-proposals: oeps/processes/oep-0054-core-contributors.rst

===========================
OEP-54: Core Contributors
===========================
.. list-table::
   :widths: 25 75

   * - OEP
     - :doc:`OEP-54 <oep-0054-core-contributors>`
   * - Title
     - Core Contributor Program
   * - Last Modified
     - 2021-12-20
   * - Authors
     - Sarina Canelake <sarina@tcril.org>
   * - Arbiter
     - Xavier Antoviaque <xavier@opencraft.com>
   * - Status
     - Draft
   * - Type
     - Process
   * - Created
     - 2021-12-20
   * - Review Period
     - TBD

.. contents::
   :local:
   :depth: 3

Abstract
========

> This document defines the Core Contributor (CC) Program, a program that grants rights to individuals

Author: Overall I agree, but the whole entire document is about Core Contributors and it's the only acronym used, so I kind of feel this acronym is ok. I could re-emphasize it at its first use in each section.  I'd like to hear what other reviewers think before committing to a change in course.

> This document defines the Core Contributor (CC) Program, a program that grants rights to individuals

Reviewer: Also agree with acronyms making things more difficult sometimes, but having read thought the document just now I did not find this CC acronym to be problematic for understanding.

Specification
=============

Core Contributor Roles
----------------------

There are many contributors to the Open edX platform that participate in a variety of ways - writing
code for the Open edX software, translating the project, performing Quality Assurance tasks, participating
in the Marketing working group, writing documentation, and more. At the time of writing we have identified
the following roles: Project Manager, Translation Lead, Code Contributors, Forum Moderators, UI/UX Designers,
Marketing Team Members, Documentation Writers, QA, and Product Managers.

All CCs are granted some Rights to the project as part of participating - the ability to
merge code, the final say on marketing copy, etc. In exchange for these Rights to contribute, we ask that
CCs abide by their Responsibilities. For more on the CC roles, and Rights & Responsibilities, see:

- `Core Contributor Role Definitions <https://openedx.atlassian.net/wiki/spaces/COMM/pages/2759460357/Core+Contributor+Role+Definitions>`_
- `Core Contributor Rights & Responsibilities <https://openedx.atlassian.net/wiki/spaces/COMM/pages/2952003698/Core+Contributor+Rights+Responsibilities+excluding+code+contributors>`_

> - `Code Committer Rights & Responsibilities <https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Rights+Responsibilities+for+Code+Contributors>`_

Author: They are core, but I don't want them defined in two places - that makes multiple places to update, high chance of getting out of sync and unclear which document is the main one. I defaulted to the wiki as the main branch because it's much easier to update the wiki (particularly, by those who don't code).

Note that roles vary in what they need from members; the wiki will be the most up to date source
of what each role entails and what types of commitments are expected.

Core Contributor Program Administration
---------------------------------------

The Core Contributor Program is administered by The Center for Re-imagining Learning (tCRIL). One or more
persons at tCRIL (the "Program administrator(s)") work with the community to make sure the program runs
smoothly: this entails facilitating communication amongst community members, `selecting new members`_,
and ensuring that existing members renew their commitment to the community and abide by the Code of Conduct.

The tCRIL administrator(s) are the point people for making sure that needed agreements are signed - this includes
a Program Agreement between the CC and tCRIL; for code contributors, an additional CLA; and for
new entities joining the community, an Entity Contributor Agreement. We are additionally asking that all employers
of CCs sign the `Revised Declaration of Commitment to the Core Contributor Program <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3216900524>`_

> to demonstrate that they are willing to engage with and commit resources to the Open edX project.

Author: Hmm. I wonder though if what a 3-person startup can offer is different from what a 30-person can? For example MITx pledges that Peter will write release notes for named releases. Should we say that's not enough?

> to demonstrate that they are willing to engage with and commit resources to the Open edX project.

Reviewer 2: Yup, I'm not saying it can't be different -- but maybe just to have a process to review and approve it? Maybe something to make part of the review process of each new core contributor?

> to demonstrate that they are willing to engage with and commit resources to the Open edX project.

Author: What about something like, "We encourage a commitment of 20h/month from each of an organization's Core Contributor, however, we recognize that circumstances differ and a different commitment may be more appropriate. Before signing the agreement, please reach out to the Program Administrator to discuss."

?

And I should probably make an email list so people can actually contact the program administrators :)

> to demonstrate that they are willing to engage with and commit resources to the Open edX project.

Reviewer 2: Sounds good to me :+1:  

Could this also be mentioned in the review of the candidature, so that can be considered as part of the review? Ie judge the skills, and also the exact commitment to the role being made?

> to demonstrate that they are willing to engage with and commit resources to the Open edX project.

Author: Yes, I can make that mention - do you mean just to consider their commitment holistically, or to make some level of commitment a requirement for being a CC?

Questions about the Program can be directed to the tCRIL administrator(s) via the ``#core-contributors`` Slack room
in the `Open edX Slack <http://openedx-slack-invite.herokuapp.com/>`_.

.. _selecting new members:

Adding New Core Contributors
----------------------------

The Open edX Core Contributor Program grants specific types of access rights to Open edX community members
which enables them to further the development and adoption of the Open edX platform. CCs earn
these rights by sustained, active participation in the Open edX community and by following the “3 Cs”:
Commitment, Conduct, Caliber. Core Contributors follow the `Code of Conduct
<https://open.edx.org/code-of-conduct/>`_, are respectful of other community members, are polite and
welcoming to all in the community, and have shown the ability to deliver high quality work over time.

> Current project CCs are always on the lookout for project contributors who demonstrate the

Author: I had written that down in my brainstorming but was unsure of it/how to go about it. I don't think it's a bad idea. What do you think would be a good way to get that feedback? The "application" should at minimum be viewable by all other CCs

> Current project CCs are always on the lookout for project contributors who demonstrate the

Reviewer 2: Maybe a forum post? Which could then be the basis for the sponsoring/review?

> Current project CCs are always on the lookout for project contributors who demonstrate the

Author: This is outdated - I rewrote the whole section, but basically my idea now is that people can just make their own post "sponsoring" themselves and the process is basically the same, although the Program Administrator would step in on a failed bid.

3 Cs and have been an overall active community member for some time. CCs, in conjunction work with the

> Program administrator(s), identify and nominate new CCs for CC roles that are ready to be expanded.

Author: ha, I think "work" is leftover from copy editing. It should read

> CCs, in conjunction with the Program administrator(s), identify...

does that make more sense?

> Program administrator(s), identify and nominate new CCs for CC roles that are ready to be expanded.

Reviewer 2: Ah, yes :) So practically, it means mentioning to you who to nominate?

> Program administrator(s), identify and nominate new CCs for CC roles that are ready to be expanded.

Author: Ah this is an outdated diff... I realized I didn't really want that, I want people to nominate whomever they like. I added a small note to the effect that if you're not sure the role is ready for expansion (for example, the UX role is not yet), you should check in before nominating.

When a new Core Contributor role is being defined and piloted, however,
nominations should proceed only once the Program Administor is confident that new Core
Contributors in that role can be supported.

New Core Contributor Nomination Process
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The process for nomination is as follows:

#. An existing CC ("sponsor") choses to nominate a community member ("candidate") for a new CC position
#. The sponsor posts a new public thread on Discourse, in the `Core Contributor

>    discussion space <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_, that

Author: Huh. That was in my notes and I'm surprised to see that caveat didn't make it in here. I'll rework this but I wanted to say that any private feedback should be sent along to the administrator, who is responsible for collecting and re-posting anon feedback publicly (in a way that doesn't connect the author to the comments). Does that make sense?

>    discussion space <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_, that

Author: Looks like this is addressed below

>    discussion space <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_, that

Author: Hmmm I didn't address

> provide a private/anonymous way to send feedback if someone wants to?

I'm so hopeful that will never be necessary, but I can think of one or two exceptional cases where that may be desired. Hm. I don't know of a great way to do this! What do you think?

>    discussion space <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_, that

Reviewer 2: What you suggested below, by gathering the feedback and then republishing it in an anonymous format, could be a good compromise?

>    discussion space <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_, that

Author: Yeah, I hope so. I can't think off the top of my head a great way to repeatably leave anonymous comments (like a google survey or another tool like that could work, but you'd have to set one up for each candidate)

   introduces the candidate, describes in a few paragraphs why they are suited to
   join the program, and provides links to examples of previous Open edX work (for example, pull requests).
   The post must mention the date when the comment period ends, which is one month from the start of the thread.
#. The sponsor posts a link to the Discourse post in the ``#core-contributors`` Slack room,
   as well as any other Slack rooms relevant to the CC role being discussed. The message must indicate
   the dates of the comment period.
#. During the comment period, other CCs should weigh in on the thread. If a CC has concerns they feel

>    cannot be raised publicly, they should consult with the Program Administrator to determine how to

Author: Yeah, how about the administrator is responsible for re-posting the feedback publicly in a way that synthesizes all the responses and makes sure the author of the feedback isn't connected to the feedback? (not sure if that phrasing makes sense - I do this with 360 review feedback, where person A writes an honest review of person B, and then I rewrite person A's feedback omitting details that would give away that A wrote the feedback)

>    cannot be raised publicly, they should consult with the Program Administrator to determine how to

Reviewer 2: Yup, that seem like a good compromise :+1:

   proceed. Remember to leave constructive criticism - if a CC candidate is not ready yet, provide some
   tips as to how they could make it into the program.
#. Over the course of the comment period, the sponsor should respond to all comments that come in. It
   may also be necessary to periodically nudge other CCs to respond to the thread.

> #. At the end of the comment period, if the candidate is approved, the candidate should work with the Program

Author: I think there needs to be some number of declarative "yes" votes, and no outstanding "no" votes. Does that feel right? How many "yes" votes make sense? How might it change with, say, 100 core contributors versus 5 product managers? I'm honestly inclined to say that after one month, 3-5 yes votes may be enough? Or should it be some % of the # of CCs in the role?

> #. At the end of the comment period, if the candidate is approved, the candidate should work with the Program

Reviewer 2: Having no outstanding "no" and at least X "yes" sounds good. It can probably make sense to have at least 5 reviews/yes as a quorum, to make sure the application did get enough attention? Not getting too many reviews also mean something, like the way some PRs don't attract reviewers? It could be reviewers from any type of roles though, to help?

> #. At the end of the comment period, if the candidate is approved, the candidate should work with the Program

Author: 5 is a good number. I do make it clear that throughout the comment period, it is the job of the candidate and the sponsor to drum up reviews.

   sponsor to help the candidate develop a plan of action so they can be a successful candidate in the future.
   
Where Do I Start?
^^^^^^^^^^^^^^^^^

New community members interested in the Core Contributor program should `get in touch with us! <https://open.edx.org/community/connect/>`_
The ``#core-contributors`` room in Slack can help guide people towards people working in their
area(s) of interest.

Existing community members who have a record of contributing to the Open edX project should feel free
to reach out to `current CCs <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3156344833/Current+Open+edX+Core+Contributors>`_
who have the role they're interested in. Engage in conversation to see what it's
like to be that type of CC; you might ask them to take a look at your work, to see if they would be
able to sponsor you for inclusion into the CC program. CCs may be pretty overloaded, so don't take
offense if the first person you reach out to doesn't have time at that moment to help. And if you don't
know where to begin, find us in ``#core-contributors`` on Slack or visit us at the next

> `Contributor's Meetup <https://calendar.google.com/calendar/embed?src=c_mvh2oito7s9n89g5nnkk49psks%40group.calendar.google.com&ctz=America%2FNew_York>`_.

Reviewer 2: The core contributors sprints could also be a good place to come get a task assigned? Or the individual workgroups meetings.

Author: Hmm. I think I want to not enshrine specific meetings in this doc, as OEPs aren't generally updated frequently, and meeting purpose/content (or the meeting itself!) may change over time. I think I'll remove the meetup calendar and point at the CC forum topics.

The Core Contributor Program is administered by The Center for Re-imagining Learning (tCRIL). One or more
persons at tCRIL (the "Program administrator(s)") work with the community to make sure the program runs
smoothly: this entails facilitating communication amongst community members, `selecting new members`_,
and ensuring that existing members renew their commitment to the community and abide by the Code of Conduct.
Program administrators are empowered to monitor Core Contributor conduct, particularly whether they follow
the "3 Cs" (Commitment, Conduct, Caliber), and may issue reminders, suspensions, and even removal from
the Program. Program administrators can be contacted at ``cc-program-admins@tcril.org``.

The tCRIL administrator(s) are the point people for making sure that needed agreements are signed - this includes
a Program Agreement between the CC and tCRIL; for code contributors, an additional CLA; and for
new entities joining the community, an Entity Contributor Agreement. We are additionally asking that all employers
of CCs sign the `Revised Declaration of Commitment to the Core Contributor Program <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3216900524>`_

> to demonstrate that they are willing to engage with and commit resources to the Open edX project. We

Reviewer 3: This is a lot of agreements. Would a list be clearer?

Should there be links to each of the agreements?

Author: The CLA is prompted when you make your first PR; you can't merge a PR until it is signed. The Entity Agreement and Program Agreement are sent to you by tCRIL when appropriate. So I don't think it is right to post any of those documents. Should I clarify how you get them?

Author: I did this clarification (of how you get each doc) and I separated the docs from the Declaration of Commitment section.

encourage a commitment of 20 hours/month from each of an organization's Core Contributors, however, we
recognize that circumstances differ and a different commitment may be more appropriate. Before signing
the agreement, please reach out to the Program Administrators to discuss.

Questions about the Program can be directed to the tCRIL administrator(s) via the ``#core-contributors`` Slack room
in the `Open edX Slack <https://openedx.slack.com/>`_.

.. _selecting new members:

Adding New Core Contributors
----------------------------

New Core Contributors are determined via a nomination process - CCs may nominate
new members, or existing community members may put themselves forward for a role.

> Current project CCs should always be on the lookout for contributors who demonstrate the

Reviewer 3: What does "project" mean in this context?

Author: Yeah that phrasing makes no sense. What about simply `Current CCs`?

"3 Cs" (Commitment, Conduct, and Caliber), follow the `Code of Conduct
<https://open.edx.org/code-of-conduct/>`_, are respectful of other community members, are polite and
welcoming to all in the community, and have been an overall active community member for some time.

> When a person like this comes to the attention of the CC, the CC should reach out and ask if

Reviewer 3: ```suggestion
When a person like this comes to the attention of a CC, the CC should reach out and ask if
```

New Core Contributor Nomination Process
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The process for nomination is as follows:

#. An existing CC ("sponsor") choses to nominate a community member ("sponsored candidate") for a new CC
   position, OR, a current community member ("organic candidate") decides to put themselves forth for nomination.

#. The sponsor or organic candidate posts a new public nomination thread on Discourse, in the `Core Contributor
   discussion space <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_.

   #. This post introduces the candidate, describes in a few paragraphs why they are suited to join the program,
      and provides links to examples of previous Open edX work (for example, pull requests).
   
   #. The post should also indicate the candidate's expected commitment to contribution - if the candidate is
      part of an organization, this should be covered as part of the organization's signature to the Declaration
      of Commitment to the Core Contributor Program, as covered in the `program administration`_ section.
   

>    #. Finally, the post must mention the date when the comment period ends, which is one month from the start of

Reviewer 4: Does this mean there's effectively a one month lag on someone becoming a CC?  Maybe that's fine; I can imagine situations where a project would really benefit from someone completing their nomination on a speedier timeline, though.  (I'm not proposing a change here, just mulling that over)

Author: Going to resolve this as we discuss in another thread

      the thread.

#. The sponsor or organic candidate posts a link to the Discourse post in the ``#core-contributors`` Slack room,
   as well as any other Slack rooms relevant to the CC role being discussed (for example, repo-specific rooms for

>    coding contributors, translation team rooms, etc). **The messages must indicate the dates of the comment period.**

Reviewer 3: Is the process the same for a contributor who wants to change or add roles?

Author: I think that makes sense. What is your opinion?

Reviewer 3: It seems like a lot of overhead for some lateral moves, but I can see an argument for defaulting to everything being consistent.

Author: Yeah - the more I think about it the more I want it to be consistent. Sure, it may be quick to go from coder to translator, but going from forum mod to coder would want the same amount of input. I could maybe see shortening the comment period one or two weeks for people changing/adding roles, pending judgement of the person nominating.

Relatedly, is a month too long of a comment period? I threw it out as a strawman - I don't think it should be any longer than that. But I also didn't want someone nominating someone on 21 Dec and accepting them on 4 Jan - no one would have time to comment.

Reviewer 4: Whoop, I'm reading from top to bottom and made a related comment above.  IMHO, yeah, sometimes a month period would feel really long.  Maybe like... two/three weeks of business days or something?  That way it's a bit resilient to holiday breaks?  Just riffing here.

Author: I have come to terms that saying "2 weeks or more, being mindful of holiday periods"

Reviewer 2: The duration seem fine to me - we can start with a duration and adapt as needed anyway. One month wasn't shocking to me though - getting a core contributor role isn't something that needs to be light or fast. This is a senior responsibility, that comes to reward months or years of consistent contribution to the platform and the community. Having a comment period that ensures a maximum of people can review it would help ensuring the quality of the core contributors being nominated.

#. During the comment period, other CCs should weigh in on the thread. As much as possible, posts should
   conclude with a definitive "yes" or "no" vote.

   #. If a CC has concerns they feel cannot be raised publicly, they should consult with the Program Administrator
      to determine how to proceed. Program administrators can be contacted at ``cc-program-admins@tcril.org``.
      The Administrator should discuss the concerns and try to figure out a way the CC could

>       post those concerns themselves; if they cannot, the Administrator should post an anonymized version of the concerns

Reviewer: ```suggestion
      post those concerns themselves; if they cannot, the Administrator should post an anonymized version of
```

>       on the Discourse thread.

Reviewer: ```suggestion
      the concerns on the Discourse thread.
```

   #. For those leaving a "no" vote: Remember to leave constructive criticism! A "no" vote should be interpreted
      as "candidate isn't ready yet". In this case, provide some tips as to how
      they could make it into the program. And please err on the side of leaving that feedback, rather than not
      replying at all.

#. Over the course of the comment period, the sponsor, sponsored candidate, or organic candidate should
   respond to all comments that come in. It may also be necessary to periodically nudge other CCs to respond
   to the thread.

   #. At any time, the candidate may choose to withdraw from the process. The candidate should leave a note
      on the thread, and close it if they wish.

#. At the end of the comment period, a candidate is approved if there are at least 5 (five) affirmative
   "yes" votes, and zero "no" votes. For roles with few existing members, it is encouraged to ask CCs of
   other roles to help evaluate the candidate.

> #. if the candidate is approved, the candidate should proactively work with the Program

Reviewer 4: ```suggestion
#. If the candidate is approved, the candidate should proactively work with the Program
```

- `Code Committer Rights & Responsibilities <https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Rights+Responsibilities+for+Code+Contributors>`_

Note that roles vary in what they need from members; the wiki will be the most up to date source
of what each role entails and what types of commitments are expected.

.. _program administration:

Core Contributor Program Administration
---------------------------------------

The Core Contributor Program is administered by The Center for Re-imagining
Learning (tCRIL). One or more persons at tCRIL (the "Program administrator(s)")
work with the community to make sure the program runs smoothly: this entails
facilitating communication amongst community members, `selecting new members`_,
and ensuring that existing members renew their commitment to the community and
abide by the Code of Conduct. Program administrators are empowered to monitor
Core Contributor conduct, particularly whether they follow the "3 Cs"
(Commitment, Conduct, Caliber), and may issue reminders, suspensions, and even

> removal from the Program. Program administrators can be contacted at
> ``cc-program-admins@tcril.org``.

Reviewer 5: Unnecessary duplication:

```suggestion
removal from the Program.
```

The tCRIL administrator(s) are the point people for making sure that needed
agreements are signed - this includes a Program Agreement between the CC and
tCRIL (sent to you by a Program Administrator); for code contributors, an

> additional CLA (signed when you make your first PR); and for new entities

Reviewer 6: Does "additional" here mean in addition to the Program Agreement, or a second CLA?

Author: "additional" here means, if you are committing code you need to also sign a CLA. Non committers do not need a CLA. I will clarify and address your other point.

joining the community, an Entity Contributor Agreement (please communicate with
a Program Administrator about this).

> We are additionally asking that all employers of CCs sign the `Revised

Reviewer 6: Mentioning employers here precludes individuals becoming CCs independent of their day job.  Are we OK with that?

Author: No, I'll clarify this point.

Author: @nedbat How would you recommend rewording this? The point is, if you employ a CC we hope you will sign the agreement. I'm not sure I agree that this wording precludes individuals.

Reviewer 6: For example, I could be employed by a company that has nothing to do with Open edX. I wouldn't expect that employer to sign a commitment. I'd be committing my own time as an individual.

Author: Ah! I see how that could be read that way! Hmm... what about,

`We are additionally asking that all companies and teams joining the Open edX community sign ...`?

Author: Ah! I see how that could be read that way! Hmm... what about,

`We are additionally asking that all companies and teams joining the Open edX community sign ...`?

Declaration of Commitment to the Core Contributor Program
<https://openedx.atlassian.net/wiki/spaces/COMM/pages/3216900524>`_ to
demonstrate that they are willing to engage with and commit resources to the
Open edX project. We encourage a commitment of 20 hours/month from each of an
organization's Core Contributors, however, we recognize that circumstances
differ and a different commitment may be more appropriate. Before signing the

> agreement, please reach out to the Program Administrators to discuss.

Reviewer 5: Can link to the suggested "Contact Program Administrators" subsection instead:

(I _think_ this is  [how you embed a named link to an internal anchor](https://jwodder.github.io/kbits/posts/rst-hyperlinks/#internal-hyperlink-targets), but I'm not sure.. RST is really hard!)

```suggestion
agreement, please `reach out to the Program Administrators <contact_admin_>`_ to discuss.
```

Questions about the Program can be directed to the tCRIL administrator(s) via
the ``#core-contributors`` Slack room in the `Open edX Slack
<https://openedx.slack.com/>`_.

.. _selecting new members:

Adding New Core Contributors
----------------------------

New Core Contributors are determined via a nomination process - CCs may nominate
new members, or existing community members may put themselves forward for a role.

Current CCs should always be on the lookout for contributors who demonstrate the
"3 Cs" (Commitment, Conduct, and Caliber), follow the `Code of Conduct
<https://open.edx.org/code-of-conduct/>`_, are respectful of other community members, are polite and

> welcoming to all in the community, and have been an overall active community member for some time.

Reviewer 6: This sentence duplicates some of the first mention of the 3 Cs.

Author: I actually intentionally duplicated this, because I am not sure how many people simply read one section of an OEP and don't click through links. Also, there is a lot on the wiki that mentions the "3 C's" and I am frequently asked what those are. I'll make this a link if you still think it is better to do so.

Reviewer 5: Duplication in code creates work to maintain, but duplication in documentation is not so bad :) I'm ok to keep this if you think it gets the point across better :)

When a person like this comes to the attention of a CC, the CC should reach out and ask if
that person would like to be nominated as a CC.

Community members who feel they fit these criteria listed above should feel empowered to
nominate themselves, as well.

Note that when a new Core Contributor role is being defined and piloted, however,
nominations should proceed only once the Program Administrator is confident that new Core

> Contributors in that role can be supported. Reach out to the Administrator when there is
> any uncertainty around the status of the role; Administrators can be contacted at ``cc-program-admins@tcril.org``.

Reviewer 5: Can link to the suggested "Contact Program Administrators" subsection instead:

```suggestion
Contributors in that role can be supported. `Reach out to the Administrator <contact_admin_>`_ when there is any uncertainty around the status of the role.
```

> 

Reviewer 5: Re @nedbat 's [comment](https://github.com/openedx/open-edx-proposals/pull/275#issuecomment-1022619102):

> Maybe it's implied here, but I think at any given time the Program Administrators will be able to say, "we have enough CCs right now, nominations are closed." True? If so, do we want to state it?

Hmm.. that's worth mentioning here, though I hope we don't ever have to invoke it! It's like saying "sorry, we're too busy to hire new people right now" :) But how about:

```suggestion
At times, the Core Contributor program may suspend accepting new members, when there is insufficient support for additional people in a given role. Program Administrators will indicate this by posting in the `Core Contributors discussion category <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_.
```

New Core Contributor Nomination Process
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The process for nomination is as follows:

> #. An existing CC ("sponsor") choses to nominate a community member ("sponsored

Reviewer 6: ```suggestion
#. An existing CC ("sponsor") chooses to nominate a community member ("sponsored
```

   candidate") for a new CC position (if they already hold a CC position, they
   should still be nominated for a new role). A current community member
   ("organic candidate") may also decide to put themselves forth for nomination.

#. The sponsor or organic candidate posts a new public nomination thread on
   Discourse, in the `Core Contributor discussion space
   <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_.

   #. This post introduces the candidate, describes in a few paragraphs why they
      are suited to join the program, and provides links to examples of previous
      Open edX work (for example, pull requests).
   
   #. The post should also indicate the candidate's expected commitment to
      contribution - if the candidate is part of an organization, this should be
      covered as part of the organization's signature to the Declaration of
      Commitment to the Core Contributor Program, as covered in the `program
      administration`_ section.
   

>    #. Finally, the post must mention the date when the comment period ends,

Reviewer 6: I don't have a better suggestion, but comments on a public thread seems like they will only be positive comments.  Practically speaking, there won't be a way to oppose a nomination, or even gently suggest that the candidate wait six months and try again.  The Python core devs have a similar public voting process (some of the topics here: https://discuss.python.org/c/committers/5).  I thought they had discussed the downside of public voting, but I can't find the discussion now.

As I say, I don't have a better suggestion.

Reviewer 6: I see now you address this below.  Thanks for considering all that.

Author: Thanks!

      which is at least 2 weeks from the start of the thread. Please be mindful
      of times such as holiday breaks where people may have limited
      availability.

#. The sponsor or organic candidate posts a link to the Discourse post in the
   ``#core-contributors`` Slack room, as well as any other Slack rooms relevant
   to the CC role being discussed (for example, repo-specific rooms for coding
   contributors, translation team rooms, etc). **The messages must indicate the
   dates of the comment period.**

#. During the comment period, other CCs should weigh in on the thread. As much
   as possible, posts should conclude with a definitive "yes" or "no" vote.

>    #. If anyone has concerns they feel cannot be raised publicly, they should
>       consult with the Program Administrators to determine how to proceed.
>       Program administrators can be contacted at
>       ``cc-program-admins@tcril.org``. The Administrators should discuss the

Reviewer 5: Can link to the suggested "Contact Program Administrators" subsection instead:
```suggestion
   #. If anyone has concerns they feel cannot be raised publicly, they should
      `consult with the Program Administrators <contact_admin_>`_ to determine how to proceed.
     The Administrators should discuss the
```

      concerns and try to figure out a way the person could post those concerns
      themselves; if they cannot, the Administrators should post an anonymized
      version of the concerns on the Discourse thread.

   #. For those leaving a "no" vote: Remember to leave constructive criticism! A
      "no" vote should be interpreted as "candidate isn't ready yet". In this
      case, provide some tips as to how they could make it into the program. And
      please err on the side of leaving that feedback, rather than not replying
      at all. A good way to deliver this feedback is "Thank you for contributing
      to the Open edX community. However, I feel you are not quite ready for
      this role, for <reasons>. Some ways you could address these are <action
      items>"

#. Over the course of the comment period, the sponsor, sponsored candidate, or

>    organic candidate should respond to all comments that come in. It may also be

Reviewer 6: Some comments won't require a response. "Joe Shmoe will make a great contributor! +1!"

Reviewer 5: ```suggestion
   organic candidate should respond to all concerns that come in. It may also be
```

   necessary to periodically nudge other CCs to respond to the thread.

   #. At any time, the candidate may choose to withdraw from the process. The
      candidate should leave a note on the thread, and close it if they wish.

#. At the end of the comment period, a candidate is approved if there are at
   least 5 (five) affirmative "yes" votes, and zero "no" votes.

   #. For candidates coming from an organization, at least one affirmative vote
      must come from a CC outside of the candidate's org. This is to ensure that
      core contributors are evaluated on their ability to communicate with the

>       community, outside of their org.

Author: I added this after conversation within tCRIL engineering. We think it's important a new CC has the approval of at least one person outside their org. Being able to communicate within your own org/team is different than being able to reach out to and communicate with someone cross-organizationally.

   #. For roles with few existing members, it is encouraged to ask CCs of other
      roles to help evaluate the candidate.

#. If the candidate is approved, the candidate should proactively work with the
   Program Administrator to get the access they need for their role. If concerns
   have been brought up, the sponsor (or Program Administrator, in the case of
   organic candidates) should discuss with the candidate what they need to work
   on to become a CC. We encourage the sponsor/Administrator to help the
   candidate develop a plan of action so that they can be successful in the
   future.
   
Where Do I Start?
^^^^^^^^^^^^^^^^^

New community members interested in the Core Contributor program should `get in touch with us! <https://open.edx.org/community/connect/>`_
First check out the `discussion forums <https://discuss.openedx.org/>`_; there are a lot of different
categories, and we encourage new members to spend some time in the forums, reading through to understand

> what's going on and jumping in to ask questions. There are many people in the `Open edX Slack <https://openedx.slack.com/>`_
> (for those joining for the first time, `here's an invite <http://openedx-slack-invite.herokuapp.com/>`_).
> The ``#core-contributors`` room in Slack can help guide people towards others working in their
> area(s) of interest.

Reviewer 5: Simplification:

```suggestion
what's going on and jumping in to ask questions. The ``#core-contributors`` room in `Open edX Slack <https://openedx.slack.com/>`_ can help guide people towards others working in their area(s) of interest. For those joining Slack for the first time, `here's an invite <http://openedx-slack-invite.herokuapp.com/>`_.
```

to reach out to `current CCs <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3156344833/Current+Open+edX+Core+Contributors>`_
who have the role they're interested in. Engaging in conversation to see what it's like to be that type of CC
is invaluable. CCs might even be able to take a look at work the community member has done, however, CCs may
be pretty overloaded, so offense shouldn't be taken if someone doesn't have time at that moment to help.

If you don't know where to begin, try `joining a working group <https://openedx.atlassian.net/wiki/spaces/COMM/pages/46793351/Working+groups>`_ -
working groups may have tasks you can pick up to start showing off your skills. Start participating on
the `discussion forums <https://discuss.openedx.org/>`_; some working groups have an active presence there,
and you'll get a chance to hone your Open edX expertise by answering questions. And finally, you can find
core contributors and ask questions about the program and your interests directly either in the
`Core Contributors discussion category <https://discuss.openedx.org/c/working-groups/core-contributors/36>`_
or in the ``#core-contributors`` room on Slack.

> 

Reviewer 5: Could we consolidate all the Program Administrator contact details down here with the rest of the contact links, and reference it where needed?

Putting this at the bottom of the page also guides people to the public channels as a first point of call, instead of sending private emails.

```suggestion
.. _contact_admin:

Contact the Program Administrators
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Questions about the Program can be directed to the tCRIL administrator(s) via the ``#core-contributors`` Slack room in the `Open edX Slack <https://openedx.slack.com/>`_ or at ``cc-program-admins@tcril.org``.
```

Author: This is a good suggestion, very much appreciated
